src/DDTrace/Integrations/Guzzle/GuzzleIntegration.php

```diff
@@ -50,14 +50,38 @@ function (SpanData $span, $args, $retval) use ($integration) {
                     $response = $retval;
                     if (\is_a($response, 'GuzzleHttp\Message\ResponseInterface')) {
                         /** @var \GuzzleHttp\Message\ResponseInterface $response */
-                        $span->meta[Tag::HTTP_STATUS_CODE] = $response->getStatusCode();
+                        $statusCode = $response->getStatusCode();
+                        $span->meta[Tag::HTTP_STATUS_CODE] = $statusCode;
+
+                        // Mark as error if status code matches configuration
+                        if (self::isClientError($statusCode)) {
+                            $span->meta[Tag::ERROR] = 1;
+                            $span->meta[Tag::ERROR_TYPE] = 'http_error';
+                            $span->meta[Tag::ERROR_MSG] = "HTTP $statusCode: " . $response->getReasonPhrase();
+                        }
```

Comment:
We should probably simplify this to `HttpClientIntegrationHelper::setClientError($span, $statusCode);`